mne/io/egi/egimff.py

```diff
@@ -686,3 +692,199 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        ch_types = egi_info['chan_type'] + egi_info['pns_types']
+    else:
+        all_data = eeg_data
+        ch_types = egi_info['chan_type']
+    all_data *= 1e-6  # convert to volts
+
+    # Load metadata into info object
+    # Exclude info['meas_date'] because record time info in
+    # averaged MFF is the time of the averaging, not true record time.
+    ch_names = [channel_naming % (i + 1) for i in
+                range(mff.num_channels['EEG'])]
+    ch_names.extend(egi_info['pns_names'])
+    info = create_info(ch_names, mff.sampling_rates['EEG'], ch_types)
+    info['nchan'] = sum(mff.num_channels.values())
+    assert info['nchan'] == all_data.shape[0], 'Number of channels does \
+                                               not match number of signals \
+                                               in binary file(s).'
+
+    # Add individual channel info
+    # Get calibration info for EEG channels
+    cals = _get_eeg_calibration_info(fname, egi_info)
+    # Initialize calibration for PNS channels, will be updated later
+    cals = np.concatenate([cals, np.repeat(1, len(egi_info['pns_names']))])
+    ch_coil = FIFF.FIFFV_COIL_EEG
+    ch_kind = FIFF.FIFFV_EEG_CH
+    chs = _create_chs(ch_names, cals, ch_coil, ch_kind, (), (), (), ())
+    chs = _read_locs(fname, chs, egi_info)
+    # Update PNS channel info
+    chs = _add_pns_channel_info(chs, egi_info, ch_names)
+    info['chs'] = chs
+
+    # Add bad channels to info
+    info['description'] = category
+    try:
+        channel_status = categories[category][0]['channelStatus']
+    except KeyError:
+        warn('Channel status data not found for condition %s. No channels \
+             will be marked as bad.' % category, category=UserWarning)
+        channel_status = None
+    bads = []
+    if channel_status:
+        for entry in channel_status:
+            if entry['exclusion'] == 'badChannels':
+                if entry['signalBin'] == 1:
+                    # Add bad EEG channels
+                    for ch in entry['channels']:
+                        bads.append(channel_naming % ch)
+                elif entry['signalBin'] == 2:
+                    # Add bad PNS channels
+                    for ch in entry['channels']:
+                        bads.append(egi_info['pns_names'][ch - 1])
+    info['bads'] = bads
+
+    # Add EEG reference to info
+    # Initialize 'custom_ref_applied' to False
+    info['custom_ref_applied'] = False
+    with mff.directory.filepointer('history') as fp:
+        history = mffpy.XML.from_file(fp)
+    for entry in history.entries:
+        if entry['method'] == 'Montage Operations Tool':
+            if 'Average Reference' in entry['settings']:
+                # Average reference has been applied
+                projector, info = setup_proj(info)
+            else:
+                # Custom reference has been applied that is not an average
+                info['custom_ref_applied'] = True
```

Comment:
A single-electrode / monopolar reference is the standard type. If this is how the data were recorded then there should be no avg ref proj and also `info['custom_ref_applied'] = False`. The `custom_ref_applied` is used for other cases, such as when the average of two electrodes is used as a reference for all channels.

Not sure if this is relevant for the "Montage Operations Tool", maybe being in this section means that it's a custom, non-average, *and non-monopolar* reference, in which case what's here makes sense.

Reply:
Typically when applying a montage operation tool, the result is something other than a monopolar reference (there is no setting to reference all electrodes to a single electrode; to do this you would have to individually map every electrode to the single electrode reference). However, it technically is possible to rereference to a single electrode, and I am not sure if there is a good way to determine this from the `history.xml`. If we can't know this info for sure, should we just always make `info['custom_ref_applied`] = False`?

Reply:
No, in that case I think the safest assumption is to set `info['custom_ref_applied'] = True` as you do here. So I think we're okay here